src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -730,6 +732,175 @@ public static nuint RotateRight(nuint value, int offset)
+        {
+            if (Sse42.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32cb (uint32_t a, uint8_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+            // Software fallback
+            return Crc32Fallback.Crc32C(crc, data);
+        }
+
+        /// <summary>
+        /// Accumulates the CRC (Cyclic redundancy check) checksum.
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The data for which to compute the checksum</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, ushort data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // unsigned int _mm_crc32_u16 (unsigned int crc, unsigned short v)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32ch (uint32_t a, uint16_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+            // Software fallback
+            return Crc32Fallback.Crc32C(crc, data);
+        }
+
+        /// <summary>
+        /// Accumulates the CRC (Cyclic redundancy check) checksum.
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The data for which to compute the checksum</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, uint data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // unsigned int _mm_crc32_u32 (unsigned int crc, unsigned int v)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32ch (uint32_t a, uint32_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+            // Software fallback
+            return Crc32Fallback.Crc32C(crc, data);
+        }
+
+        /// <summary>
+        /// Accumulates the CRC (Cyclic redundancy check) checksum.
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The data for which to compute the checksum</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, ulong data)
+        {
+            if (Sse42.X64.IsSupported)
+            {
+                return (uint)Sse42.X64.Crc32(crc, data);
+            }
+            if (Sse42.IsSupported)
+            {
+                return Sse42.Crc32(Sse42.Crc32(crc, (uint)data), (uint)(data >> 32));
```

Comment:
More readable when broken apart:
```suggestion
                uint result = Sse42.Crc32(crc, (uint)(data));
                return Sse42.Crc32(result, (uint)(data >> 32));
```